### PR TITLE
Streamlit demo and changes to model.py

### PR DESCRIPTION
**Pull request by meetnaren**

### CODE REVIEW

File: ner/model.py [hunk 1]
```diff
@@ -1,10 +1,29 @@
...
+import os
+import spacy
 
-from ner import settings
-from ner.db import DB
+#from ner import settings
+#from ner.db import DB
 
-print(settings.env)
+#print(settings.env)
```
Comment (happilyeverafter95) [inline ner/model.py:10]: can we remove these comments?

File: ner/model.py [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
+
+def extract_entities(input_text):
+    # get latest model
+    models_dir = os.path.join('..', 'Models') if __name__ == '__main__' else 'Models'
```
Comment (happilyeverafter95) [inline ner/model.py:14]: we need better management around the base path. this is typically managed as an attribute in the `settings` class.

File: ner/model.py [hunk 3]
```diff
@@ -1,10 +1,29 @@
...
+    latest_model = max([d for d in os.listdir(models_dir)])
```
Comment (happilyeverafter95) [inline ner/model.py:15]: this is the same as
```suggestion
    latest_model = max(os.listdir(models_dir))
```
can you remind me again how the models are named? and is this a reliable way of picking the latest model?

File: ner/model.py [hunk 4]
```diff
@@ -1,10 +1,29 @@
...
+    latest_model_path = os.path.join(models_dir, latest_model)
+    nlp = spacy.load(latest_model_path)
+    print(f'Loaded model {latest_model}')
```
Comment (happilyeverafter95) [inline ner/model.py:18]: can we switching to logging?

File: ner/model.py [hunk 5]
```diff
@@ -1,10 +1,29 @@
...
+
+    # apply latest model on input text
```
Comment (happilyeverafter95) [inline ner/model.py:20]: I think your code is clean enough that you don't need a lot of these comments

File: ner/model.py [hunk 6]
```diff
@@ -1,10 +1,29 @@
...
+    doc = nlp(input_text)
+    print(doc)
+
+    # return the entities and texts
+    result = [(ent.label_, ent.text) for ent in doc.ents]
```
Comment (happilyeverafter95) [inline ner/model.py:25]: I would just return `[(ent.label_, ent.text) for ent in doc.ents] `

File: ner/model.py [hunk 7]
```diff
@@ -15,3 +34,6 @@ def load_data(sql):
 
     def placeholder_method(self, text):
         return text
+
+if __name__ == '__main__':
+    print(extract_entities('My name is Narendran Santhanam'))
```
Comment (happilyeverafter95) [inline ner/model.py:39]: What is this for? :) can we move it elsewhere?

File: ner/model.py [hunk 8]
```diff
@@ -15,3 +34,6 @@ def load_data(sql):
 
     def placeholder_method(self, text):
```
Comment (happilyeverafter95) [inline ner/model.py:35]: you can probably get rid of this in favor of `extract_entities`

File: demo.py [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import streamlit as st
+from ner.model import extract_entities
+
+txt = st.text_area('Text to analyze')
+
+result = extract_entities(txt)
+
+st.write('Entities recognized:')
+result_dict = {}
+if result:
```
Comment (happilyeverafter95) [inline demo.py:10]: isn't this condition redundant?

Comment (happilyeverafter95) [inline demo.py:10]: my personal preference is to be explicit with your intentions and check for `result == []`, but this is fine

File: ner/model.py (hunk 3 above)
Comment (meetnaren) [reply ner/model.py:15]: The models are named thus: 'model_YYYYMMDD-HHMMSS', so I think it should work?

File: ner/model.py (hunk 7 above)
Comment (meetnaren) [reply ner/model.py:39]: !!! I have no idea why I didn't see that before pushing it ***cringes***

File: ner/model.py (hunk 3 above)
Comment (happilyeverafter95) [reply ner/model.py:15]: yes, that should work. My small suggestion still stands since the output of `listdir` is a list of a strings.

**Pull request closed without merging**